src/main/webapp/app/shared/pipes/shared-pipes.module.ts

```diff
@@ -13,6 +13,7 @@ import { ExerciseCourseTitlePipe } from 'app/shared/pipes/exercise-course-title.
 import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
 import { GradeStepBoundsPipe } from 'app/shared/pipes/grade-step-bounds.pipe';
 import { NegatedTypeCheckPipe } from 'app/shared/pipes/negated-type-check.pipe';
+import { UnreadMessagesPipe } from './unread-messages.pipe';
```

Comment:
Why do you create a pipe for that and not only a function in the corresponding .ts file?

Reply:
The problem is related to the change detection, where angular uses to  detect the changed data and update the DOM's affected parts, even in mouse move. So the function was being called every time a change detection was detected.

Reply:
In that case it's ok. The only other option would be to calculate and store these values once e.g. in onInit